source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -478,6 +484,12 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
                 if (bindInputValue)
                 {
                     inputSocket->setValue(bindInputValue);
+
+                    input->setIsBindInput(true);
+                    ColorManagementSystemPtr colorManagementSystem = context.getShaderGenerator().getColorManagementSystem();
+                    const string& targetColorSpace = context.getOptions().targetColorSpaceOverride.empty() ?
+                        element->getDocument()->getActiveColorSpace() : context.getOptions().targetColorSpaceOverride;
+                    graph->populateInputColorTransformMap(colorManagementSystem, graph->_nodeMap[newNodeName], bindInput, targetColorSpace);
                 }
                 inputSocket->setPath(bindInput->getNamePath());
                 input->setPath(inputSocket->getPath());
```

Comment:
At line 486, you'll note that a check for a connection is done. If there is no connection then you can add in a transform. At line 498 it's adding in the link your probably seeing. This is to connect this to to an "output socket" which is of type ShaderGraphOutputSocket.

Reply:
Above at line 448, is the check for BindParam. In this case it's connected to a ShaderGraphInputSocket which could be  as a test guard as well. Both basically mean your not connected to an internal node.